src/Microsoft.Azure.WebJobs/BlobAttribute.cs

```diff
@@ -77,6 +76,7 @@ public BlobAttribute(string blobPath, FileAccess access)
 
         /// <summary>Gets the path of the blob to which to bind.</summary>
         [AutoResolve]
+        [BlobNameValidation]
```

Comment:
>[BlobNameValidation] [](start = 8, length = 20)

This is a huge step forward. The error messages now come from the DataAnnotation system. This lets us remove some explicit error checking; and it also unifies the various %%, {} cases.